rpcs3/Emu/Io/PadHandler.cpp

```diff
@@ -701,8 +703,12 @@ void PadHandlerBase::process()
 			if (!last_connection_status[i])
 			{
 				input_log.success("%s device %d connected", m_type, i);
-				pad->m_port_status |= CELL_PAD_STATUS_CONNECTED;
-				pad->m_port_status |= CELL_PAD_STATUS_ASSIGN_CHANGES;
+				atomic_storage<u32>::fetch_or(pad->m_port_status, CELL_PAD_STATUS_CONNECTED + CELL_PAD_STATUS_ASSIGN_CHANGES);
+
+				std::array<s32, CELL_MAX_PADS> status = fill_array(-1);
+				status[i] = CELL_PAD_STATUS_CONNECTED;
+				cell_PadEnqueueConnectionChanges(status);
```

Comment:
Just make a helper function that only takes an index and 0 or CELL_PAD_STATUS_CONNECTED.
The pad Händler shouldn't deal with these cell intrinsics